easybuild/easyconfigs/a/Amber/Amber-12-ictce-4.1.13.patch

```diff
@@ -0,0 +1,130 @@
++         ld=' mpiifort '
+       else
+          ld=' ifort '
+       fi
+@@ -1029,7 +1032,7 @@
+         fi
+ 
+         if [ "$mpi" = 'yes' ]; then
+-          mpi_inc=`(mpicc -show 2>&1) | awk 'BEGIN{i=0} {while (i < NF) {if ( substr($i, 1, 2) == "-I" ) {printf("%s ", $i);}; i++;}}'`
++          mpi_inc=`(mpiicc -show 2>&1) | awk 'BEGIN{i=0} {while (i < NF) {if ( substr($i, 1, 2) == "-I" ) {printf("%s ", $i);}; i++;}}'`
+           pmemd_cu_includes="$pmemd_cu_includes $mpi_inc"
+           pmemd_cu_defines="$pmemd_cu_defines -DMPI -DMPICH_IGNORE_CXX_SEEK"
+           pmemd_coptflags="$pmemd_coptflags -DMPICH_IGNORE_CXX_SEEK"
+@@ -1581,17 +1584,17 @@
+     blas=skip
+     flibs="-larpack "
+     flibsf="-larpack "
+-    mkll="$MKL_HOME/lib/32"
++    mkll="$MKL_HOME/mkl/lib/32"
+     mkl_processor="32"
+     mkl_procstring="ia32"
+     mklinterfacelayer='libmkl_intel.a'
+     if [ "$x86_64" = 'yes' ]; then
+-        if [ -d "$MKL_HOME/lib/em64t" ]; then
+-            mkll="$MKL_HOME/lib/em64t"
++        if [ -d "$MKL_HOME/mkl/lib/em64t" ]; then
++            mkll="$MKL_HOME/mkl/lib/em64t"
+             mkl_processor="em64t"
+             mkl_procstring="em64t"
+         else
+-            mkll="$MKL_HOME/lib/intel64"
++            mkll="$MKL_HOME/mkl/lib/intel64"
+             mkl_processor="intel64"
+             mkl_procstring="intel64"
+         fi
+@@ -1599,7 +1602,7 @@
+         mklinterfacelayer='libmkl_intel_lp64.a'
+     fi
+     if [ "$processor" = "ia64" ]; then
+-        mkll="$MKL_HOME/lib/64"
++        mkll="$MKL_HOME/mkl/lib/64"
+         mkl_processor="64"
+         mkl_procstring="ia64"
+         mklinterfacelayer='libmkl_intel_lp64.a'
+@@ -2354,12 +2357,14 @@
+   if [ "$crayxt5" = 'yes' -o "$compiler" = 'cray' ]; then
+       cc="cc"
+       fc="ftn"
+-      mpi_flag="-DMPI "
++  elif [ "$compiler" = 'intel' ]; then
++      cc="mpiicc"
++      fc="mpiifort"
+   else
+       cc="mpicc"
+       fc="mpif90"
+-      mpi_flag="-DMPI "
+   fi
++  mpi_flag="-DMPI "
+ fi
+ 
+ #------------------------------------------------------------------------------
+diff -ur unpatched/AmberTools/src/cpptraj/configure amber12/AmberTools/src/cpptraj/configure
+--- unpatched/AmberTools/src/cpptraj/configure	2013-04-13 00:04:14.000000000 +1200
++++ amber12/AmberTools/src/cpptraj/configure	2014-10-03 16:34:21.961264423 +1300
+@@ -35,7 +35,7 @@
+   echo "  EXPERIMENTAL OPTIONS:"
+   echo "    -profile   : Use Gnu compiler profiling (>= V4.5)"
+   echo "    -gprofile  : Use Gnu compiler GLIBC profiling (>= V4.5)"
+-  echo "    -mpi       : Use mpicc/mpicxx to compile."
++  echo "    -mpi       : Use MPI compilers to compile."
+   echo "    NOTE: -openmp and -mpi are mutually exclusive, as are -profile and -gprofile."
+ #  echo "   --with-hdf5=<DIR>"
+ }
+@@ -560,9 +560,15 @@
+   echo "Using MPI"
+   DIRECTIVES="$DIRECTIVES -DMPI"
+   if [[ $USECRAY -eq 0 ]] ; then
+-    CC=mpicc
+-    CXX=mpicxx
+-    FC=mpif90
++    if [[ "${CC}" == "icc" ]] ; then
```

Comment:
if you enable the `usempi` toolchain option in the easyconfig, you can just use `CC=${CC}` here (since `$CC` will be `mpicc` then, and `I_MPI_CC` will be set to `icc`, which is equivalent to use `mpiicc`), rather than having a check on `${CC}` and then hard setting the MPI wrapper names yourself